harness/go/harness/util.go

```diff
@@ -238,7 +238,11 @@ func LoadTLSConfig(clientCertPath, clientKeyPath string) (*tls.Config, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to load certs: %s", err)
 		}
-		return &tls.Config{Certificates: []tls.Certificate{cert}}, nil
+		tlsConfig := &tls.Config{Certificates: []tls.Certificate{cert}}
+		if tlsServerName != "" {
+			tlsConfig.ServerName = tlsServerName
+		}
```

Comment:
```suggestion
		tlsConfig := &tls.Config{Certificates: []tls.Certificate{cert}, ServerName: tlsServerName}
```

No need for the conditional